contrib/babelfishpg_tds/src/backend/tds/tdsbulkload.c

```diff
@@ -102,6 +103,12 @@ do \
 		FetchMoreBcpPlpData(message, dataLen); \
 } while(0)
 
+void
+TdsResetBcpOffset()
+{
+	offset = 0;
```

Comment:
Can't we reset this offset whenever we call GetBulkLoadRequest()?

Do we expect a non-zero offset across two GetBulkLoadRequest() call?

Besides, if you're using this variable in catch block, better to declare it as volatile.